command/agent/consul/client.go

```diff
@@ -807,6 +813,117 @@ func (c *ServiceClient) checkRegs(ops *operations, serviceID string, service *st
 	return checkIDs, nil
 }
 
+//TODO(schmichael) remove
+type noopRestarter struct{}
+
+func (noopRestarter) Restart(context.Context, *structs.TaskEvent, bool) error { return nil }
+
+// makeAllocTaskServices creates a TaskServices struct for a group service.
+//
+//TODO(schmichael) rename TaskServices and refactor this into a New method
```

Comment:
Yeah, the tricky thing is that we don't have any facility for just registering checks, and re-registering the service+checks (via TaskService struct) could end up registering a new service entry with a different ID & Port.

I don't think there's a reason we couldn't add check register/update/remove methods, but there may be better ways. We could register HTTP/gRPC checks along with the group service, but for script checks we need the driver handle (implements the ScriptExec interface). We already mess that up handle/exec handling in some circumstances (part of #6110), so maybe a new mechanism just for it would be generally useful?